k8s/operators/src/pool/diskpool/v1beta1.rs

```diff
@@ -107,6 +107,7 @@ impl DiskPoolStatus {
     /// Set when Pool is not found for some reason.
     pub fn not_found() -> Self {
         Self {
+            cr_state: CrPoolState::Created,
```

Comment:
What if the pool is not created yet?
Should we check what the current cr_state is and keep ip?

Reply:
`match dsp.status {
        Some(DiskPoolStatus {
            cr_state: CrPoolState::Creating,
            ..
        }) => dsp.create_or_import().await,
        Some(DiskPoolStatus {
            cr_state: CrPoolState::Created,
            ..
        })
        | Some(DiskPoolStatus {
            cr_state: CrPoolState::Terminating,
            ..
        }) => dsp.pool_check().await,
        // We use this state to indicate its a new CRD however, we could (and
        // perhaps should) use the finalizer callback.
        None => dsp.init_cr().await,
    }`

Reply:
If pool was never created it will be in CrPoolState::Creating state. If no field at all for the new create event its `init_cr()`

Reply:
checking if deletion timestamp is set. If yes, Will mark it as Teminating

Reply:
Seems overcomplicated, not not simply pass self to not_found and have not found keep existing cr states and clear status?

Reply:
Done. We are passsing current state now.